### PR TITLE
refactor: Extract tree composable

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Extract the tree logic from the dependencies screen into a separate composable to make it reusable.

### CODE REVIEW

File: src/main/kotlin/ui/dependencies/Dependencies.kt [hunk 1]
```diff
@@ -159,14 +111,15 @@ fun Dependencies(viewModel: DependenciesViewModel) {
...
                                 Column {
-                                    when (item) {
-                                        is DependencyTreeProject -> ProjectDetails(item)
-                                        is DependencyTreeScope -> ScopeDetails(item)
-                                        is DependencyTreePackage -> PackageDetails(item)
+                                    when (item.node.value) {
+                                        is DependencyTreeProject -> ProjectDetails(item.node.value)
+                                        is DependencyTreeScope -> ScopeDetails(item.node.value)
+                                        is DependencyTreePackage -> PackageDetails(item.node.value)
+                                        is DependencyTreeError -> ErrorDetails(item.node.value)
```
Comment (sschuberth) [inline src/main/kotlin/ui/dependencies/Dependencies.kt:122]: When looking at added lines like this, the change seems to do a bit than just extracting existing logic, right?

File: src/main/kotlin/utils/MaterialIcon.kt [hunk 2]
```diff
@@ -6,6 +6,7 @@ enum class MaterialIcon(val resource: String) {
     ASSESSMENT("material/assessment.svg"),
     BOOKMARK_BORDER("material/bookmark_border.svg"),
     BUG_REPORT("material/bug_report.svg"),
+    CHEVRON_RIGHT("material/chevron_right.svg"),
```
Comment (sschuberth) [inline src/main/kotlin/utils/MaterialIcon.kt:9]: Similar here, this added resource indicates to me that more than just pure refactoring is going on.

**Review by sschuberth — APPROVED** [review]

**Review by sschuberth — APPROVED** [review]
Approving despite my comments.

File: src/main/kotlin/ui/dependencies/Dependencies.kt (hunk 1 above)
Comment (mnonnenmacher) [reply src/main/kotlin/ui/dependencies/Dependencies.kt:122]: I had to make some adaptions to make the tree more generic but there were no major changes.

File: src/main/kotlin/utils/MaterialIcon.kt (hunk 2 above)
Comment (mnonnenmacher) [reply src/main/kotlin/utils/MaterialIcon.kt:9]: Yes, I think you found the two relevant changes, I have mentioned them in the commit message now.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**